app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/service/TrackerBlockingVpnService.kt

```diff
@@ -232,8 +231,10 @@ class TrackerBlockingVpnService : VpnService(), CoroutineScope by MainScope() {
                 return@withContext
             }
 
-            // We need to rethink how to log this state. This will likely change.
-            vpnServiceStateStatsDao.insert(VpnServiceStateStats(state = ENABLING))
+            vpnServiceCallbacksPluginPoint.getPlugins().forEach {
+                logcat { "VPN log: onVpnStarting ${it.javaClass} callback" }
+                it.onVpnStarting(this)
```

Comment:
Since `it.onVpnStarting(this)` now changes the state within a coroutine, isn't it now possible that another thread enters this critical section and the sate hasn't yet changed to `ENABLING`? It's probably ok for now, but I think it's still a race condition and we need to further re-think how we keep track of states.

Reply:
In principle I think that was also the case before. `Dispatchers.IO` is backed by a thread pool. 
But yeah, I think we need to about, also because we should not use `Dispatchers.IO` anymore.